webhook/webhook.py

```diff
@@ -1,18 +1,41 @@
 """Contain all the logic for handling webhooks in a class."""
 import logging
+import hmac
+import hashlib
 from model.user import User
 from model.team import Team
 from db.facade import DBFacade
 from typing import Dict, Any, cast, List
 from command import ResponseTuple
+from config import Credentials
 
 
 class WebhookHandler:
     """Encapsulate methods for GitHub webhook triggered events."""
 
-    def __init__(self, db_facade: DBFacade) -> None:
+    def __init__(self, db_facade: DBFacade, credentials: Credentials) -> None:
         """Give handler access to the database."""
         self.__facade = db_facade
+        self.__secret = credentials.github_webhook_secret
+
+    def verify_hash(self, request_body: bytes, xhub_signature: str):
+        """
+        Verify if a webhook event comes from GitHub.
+
+        :param request_body: Byte string of the request body
+        :param xhub_signature: Hashed signature to validate
+        :return: Return True if the signature is valid, False otherwise
+        """
+        h = hmac.new(bytes(self.__secret, encoding='utf8'),
+                     request_body, hashlib.sha1)
+        verified = hmac.compare_digest(
+            bytes("sha1=" + h.hexdigest(), encoding='utf8'),
+            bytes(xhub_signature, encoding='utf8'))
+        if verified:
+            logging.info("Webhook signature verified")
+        else:
+            logging.warning("Webhook signature not verified")

```

Comment:
I think it would be valuable to specify that this is (supposed to be) a _Github_ webhook. Also, it might be useful to include some more details about it - e.g. maybe include the signature that failed to verify?

Reply:
and done?